c7n/resources/awslambda.py

```diff
@@ -113,6 +150,25 @@ def _augment(r):
 @AWSLambda.action_registry.register('delete')
 class Delete(BaseAction):
     """Delete a lambda function (including aliases and older versions).
+
+    It is recommended to apply a filter to the delete function to avoid
+    deletion of all lambda functions.
+
+    :example:
+
+        .. code-block: yaml
+
+            policies:
+              - name: lambda-iot-delete
+                resource: lambda
+                filters:
+                  - type: event-source
```

Comment:
yes re value filter subclass for rich comparisons. no re exposed syntax/params of value filter